cpu/sam0_common/periph/adc.c

```diff
@@ -102,8 +102,8 @@ static int _adc_configure(adc_res_t res)
     /* Power On */
     PM->APBCMASK.reg |= PM_APBCMASK_ADC;
     /* GCLK Setup */
-    GCLK->CLKCTRL.reg = (uint32_t)(GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0 |
```

Comment:
Why not just keep this line and the defines in the board periph, and replace `GCLK_CLKCTRL_GEN_GCLK0` by `ADC_0_CLK_SOURCE` here ? (Maybe ADC_0_CLK_SOURCE could also be renamed in GCLK_SOURCE)

I think it's the simplest: one can easily override the default GCLK source in the board conf and it avoids having to introduce another struct.

Reply:
@aabadie Then we would have to declare ONDEMAND & RUNSTDBY as #define within the periph_conf.h file later and keep the current ADC channels struct as it ?